Content.Server/GameObjects/Components/Atmos/BreathMaskComponent.cs

```diff
@@ -0,0 +1,100 @@
+﻿using System.Diagnostics;
+using Content.Server.GameObjects.Components.GUI;
+using Content.Server.GameObjects.Components.Items.Storage;
+using Content.Shared.GameObjects;
+using Content.Shared.GameObjects.Components.Inventory;
+using Content.Shared.GameObjects.Verbs;
+using Content.Shared.Interfaces;
+using Content.Shared.Interfaces.GameObjects.Components;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Interfaces.GameObjects;
+using Robust.Shared.Localization;
+using Serilog;
+
+namespace Content.Server.GameObjects.Components.Atmos
+{
+    [RegisterComponent]
+    public class BreathMaskComponent : ItemComponent, IConnectableToGasTank
+    {
+        private InventoryComponent _connectedInventorySystem;
+        private GasTankComponent ConnectableGasTank
+        {
+            get
+            {
+                if (_connectedInventorySystem == null) return null;
+                return _connectedInventorySystem.TryGetSlotItem<GasTankComponent>(EquipmentSlotDefines.Slots.BACKPACK,
+                    out var tank) ? tank : null;
+            }
+        }
```

Comment:
@Zumorica I am not sure what would be the best way to choose desired tank to connect. Maybe instead of connecting tank to mask/suit/whatever can be used for breathing it makes sense to change the logic to connect tank instead?